api/src/main/proto/bio/terra/tanagra/underlay/entities.proto

```diff
@@ -0,0 +1,36 @@
+// Protobuf definitions for an underlay's logical entities and their relationships.
+
+syntax = "proto3";
+
+package bio.terra.tanagra.underlay;
+option java_package = "bio.terra.tanagra.proto.underlay";
+option java_multiple_files = true;
+
+import "bio/terra/tanagra/underlay/shared.proto";
+
+
+// A logical entity within an underlay.
+message Entity {
+  // The name of the entity.
+  string name = 1;
+  // The attributes in an entity.
+  repeated Attribute attributes = 2;
+}
+
+// An attribute within an entity.
+message Attribute {
+  // The name of the attribute.
+  string name = 1;
+  // The type of data for the attribute.
+  DataType data_type = 2;
+}
+
+// A relationship between 2 entities.
+message Relationship {
+  // The name of the relationship.
+  string name = 1;
+  // The name of the first entity.
+  string entity1 = 2;
```

Comment:
Is the order of entity1 vs entity2 meaningful, e.g. directional? Do we presume many:many here, or do we not care?

If the order matters, may want to consider renaming

Reply:
So far, I'm not sure that we care. Subject to change; I could imagine us having a RelationshipType { one_to_one, one_to_many, many_to_many}.